src/index.js

```diff
@@ -1,75 +1,38 @@
 'use strict';
 
-/* eslint no-shadow: 0 */
-/* eslint no-else-return: 0 */
-/* eslint new-cap: 0 */
-/* eslint prefer-const: 0 */
-
 const mongoose = require( 'mongoose' );
 const conn = mongoose.connection;
 const Permissions = require( './schemas/permissionSchema.js' );
 mongoose.Promise = Promise;
 
-module.exports.connect = function mongoConnect() {
+const connect = () => {
     return new Promise(( resolve, reject ) => {
-        // check to see if we're already connected
-        if ( conn.readyState === 1 ) {
-            // if so, begin
-            resolve();
-        }
-        // if not, connect
-        else {
-            let mongoAddress;
-            if ( process.env.NODE_ENV === 'development' ) {
-                mongoAddress = 'mongodb://' + process.env.IP + ':27017';
-            }
-            else {
-                mongoAddress = 'mongodb://localhost:27017';
-            }
+        // Already connected?
+        if ( conn.readyState === 1 ) return resolve();
 
-            mongoose.connect( mongoAddress );
-            conn.on( 'error', ( err ) => {
-                reject( err );
-            });
-
-            conn.on( 'connected', () => {
-                resolve();
-            });
-            // if we get this far and nothing has happened, we assume mongo has not started
-        }
+        // if not, connect
+        const IP = process.env.NODE_ENV === 'development' ? process.env.ip : 'localhost';
+        mongoose.connect( `mongodb://${IP}:27017` );
+        conn.on( 'error', ( err ) => reject( err ));
+        conn.on( 'connected', resolve );
```

Comment:
See https://github.com/Brinkbit/the-brink-auth/blob/master/src/config.js for an example of a full docker-friendly mongo configuration